File: csharp/algorithms/AppleAndOrange.cs
using System;
using System.IO;
using run_hack;

// Solution for Apple and Orange https://www.hackerrank.com/challenges/apple-and-orange/problem
namespace algorithms
{
  namespace implementation
  {
    public class AppleAndOrange : IHackerRankSolution
    {
      private static readonly int S_AND_APPLE_TREE = 0;
      private static readonly int T_AND_ORANGE_TREE = 1;
      private int[] _sAndT;
      private int[] _trees;
      private int[] _apples;
      private int[] _oranges;

      public void PrepareData(string[] data)
      {
        this._sAndT = Array.ConvertAll(data[0].Split(' '), Convert.ToInt32);
        
        this._trees = Array.ConvertAll(data[1].Split(' '), Convert.ToInt32);
        
        this._apples = Array.ConvertAll(data[3].Split(' '), Convert.ToInt32);
        this._oranges = Array.ConvertAll(data[4].Split(' '), Convert.ToInt32);
      }

      public void RunSolution()
      {
        var hits = new int[2] {0, 0};
        foreach (var apple in this._apples)
        {
          hits[0] += this.DidHit(apple + this._trees[S_AND_APPLE_TREE]);
        }
        foreach (var orange in this._oranges)
        {
          hits[1] += this.DidHit(orange + this._trees[T_AND_ORANGE_TREE]);
        }

        foreach (var hit in hits)
        {
          Console.WriteLine(hit);
        }
      }

      private int DidHit(int distance)
      {
        return (distance >= this._sAndT[S_AND_APPLE_TREE] && distance <= _sAndT[T_AND_ORANGE_TREE])
          ? 1
          : 0;
      }
    }
  }
}

File: java/src/com/hackerrank/algorithms/impl/AppleAndOrange.java
package com.hackerrank.algorithms.impl;

import com.hackerrank.main.HackerRankSolution;

import java.io.BufferedReader;
import java.io.IOException;
import java.util.Arrays;
import java.util.stream.Stream;

// Solution for https://www.hackerrank.com/challenges/apple-and-orange/problem
public class AppleAndOrange implements HackerRankSolution {
  private static final Integer S_AND_APPLE_TREE = 0;
  private static final Integer T_AND_ORANG_TREE = 1;
  private int[] sAndT, trees, apples, oranges;
  private final int[] hits = {0, 0};
  @Override
  public void prepareData(BufferedReader testFile) {
    try {
      this.sAndT = Stream.of(testFile.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
      this.trees = Stream.of(testFile.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
      int[] unused = Stream.of(testFile.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
      this.apples = Stream.of(testFile.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
      this.oranges = Stream.of(testFile.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
    } catch (IOException e) {
      e.printStackTrace();
    }
    //Contains array sizes which are not really needed
  }

  @Override
  public void runSolution() {
    Arrays.stream(apples).forEach(apple -> this.hits[0] += didHit(trees[S_AND_APPLE_TREE] + apple));
    Arrays.stream(oranges).forEach(orange -> this.hits[1] += didHit(trees[T_AND_ORANG_TREE] + orange));
  }

  public void printResults() {
    Arrays.stream(this.hits).forEach(System.out::println);
  }

  private Integer didHit(Integer distance) {
    return distance >= this.sAndT[S_AND_APPLE_TREE] && distance <= this.sAndT[T_AND_ORANG_TREE] ? 1 : 0;
  }
}
